pkg/asset/installconfig/gcp/validation.go

```diff
@@ -462,6 +465,26 @@ func validateNetworks(client API, ic *types.InstallConfig, fieldPath *field.Path
 	return allErrs
 }
 
+func validateServiceEndpoints(_ API, ic *types.InstallConfig, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// attempt to resolve all the custom (overridden) endpoints. If any are not reachable,
+	// then the installation should fail not skip the endpoint use.
+	for id, serviceEndpoint := range ic.GCP.ServiceEndpoints {
+		// Ignore e2e.local for unit tests.
+		if serviceEndpoint.URL == "e2e.local" {
+			logrus.Warnf("Skipping service endpoint validation for %s", serviceEndpoint.Name)
+			continue
+		}
+		if _, err := url.Parse(serviceEndpoint.URL); err != nil {
+			allErrs = append(allErrs, field.Invalid(fieldPath.Child("serviceEndpoint").Index(id), serviceEndpoint.URL, err.Error()))
+		} else if _, err := http.Head(serviceEndpoint.URL); err != nil {
+			allErrs = append(allErrs, field.Invalid(fieldPath.Child("serviceEndpoint").Index(id), serviceEndpoint.URL, err.Error()))
```

Comment:
I think we probably want to wrap this error message with something like "error when connecting to endpoint:"
```suggestion
			allErrs = append(allErrs, field.Invalid(fieldPath.Child("serviceEndpoint").Index(id), serviceEndpoint.URL, fmt.Errorf("error when connecting to endpoint: %w", err.Error())))
```

Reply:
+1